docs/command_reference.md

```diff
@@ -0,0 +1,297 @@
+
+Users can sign content into a notary trusted collection by running:
+```bash
+$ notary add -p <GUN> <target_name> <target_file>
+```
+
+In the above command, the `<target_name>` corresponds to the name we want to associate the `<target_file>` with in the trusted collection. Notary will sign the hash of the `<target_file>` into its trusted collection.
+Instead of adding a target by file, you can specify a hash and byte size directly:
+```bash
+$ notary addhash -p <GUN> <target_name> <byte_size> --sha256 <sha256Hash>
+```
+
+To check that your trust data was published successfully to the notary server, you can run:
+```bash
+$ notary list <GUN>
+```
+
+To remove targets from a trusted collection, you can run:
+```bash
+$ notary remove -p <GUN> <target_name>
+```
+
+## Delete trust data
+
+Users can remove all notary signed data for a trusted collection by running:
+
+```bash
+$ notary delete <GUN> --remote
+```
+
+If you don't include the `--remote` flag, Notary deletes local cached content
+but will not delete data from the Notary server.
+
+## Change the passphrase for a key
+
+The Notary CLI client manages the keys used to sign the trusted collection. These keys are encrypted at rest.
+To list all the keys managed by the Notary CLI client, run:
+
+```bash
+$ notary key list
+```
+
+To change the passphrase used to encrypt one of the keys, run:
+
+```bash
+$ notary key passwd <key_id>
+```
+
+## Rotate keys
+
+If one of the private keys is compromised you can rotate that key, so that
+content that was signed with those keys stop being trusted.
+
+For keys that are kept offline and managed by the Notary CLI client, such the
+keys with the root, targets, and snapshot roles, you can rotate them with:
+
+```bash
+$ notary key rotate <GUN> <key_role>
+```
+
+The Notary CLI client generates a new key for the role you specified, and
+prompts you for a passphrase to encrypt it.
+Then you're prompted for the passphrase for the key you're rotating, and if it
+is correct, the Notary CLI client immediately contacts the Notary server to auto-publish the
+change, no `-p` flag needed.
+
+After a rotation, all previously existing keys for the specified role are replaced with the new key.
+
+You can also rotate keys that are stored in the Notary server, such as the keys
+with the snapshot or timestamp role. To do this, use the `-r` flag:
+
+```bash
+$ notary key rotate <GUN> <key_role> -r
+```
+
+## Importing and exporting keys
+
+Notary can import keys that are already in a PEM format:
+```bash
+$ notary key import <pemfile> --role <key_role> --gun <key_gun>
+```
+
+The `--role` and `--gun` flags can be provided to specify a role and GUN to import the key to, which overrides any information contained in PEM headers.
+Note that for root and delegation keys, the `--gun` flag is ignored because these keys can be shared across GUNs.
+If no `--role` or `--gun` is given, notary will assume that the key is to be used for a delegation role.
```

Comment:
Might want to note that it will be available for signing regardless of role, but will be shown as a delegation key in commands such as `notary key list`